### PR TITLE
feat(function): enable fetch content from a private repository

### LINKED ISSUE #3: Does not able to get file from private repositories
# Overview
The action could not get configurations from private repositories

# How to Reproduce

1. Create a sample procheck config in a private repository
2. Run the action with the configPath directing to the config in a private repository

# Output

It won't be able to read config variable because it cannot read the configuration

`TypeError: configs is not iterable`
image

### PR DESCRIPTION
**Pull request by dennypradipta**
This PR solves #3 which is to handle fetch config from a private repository. In order to do that, we
need one more Github Actions parameters called `githubToken` to insert user Github PAT and add them into the
headers.

fix #3

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -8,6 +8,9 @@ inputs:
   configPath:
     required: true
     description: 'Configs path'
+  token:
```
Comment (kevin-hyperjump) [inline action.yml]: Rather than use `token`, I think better to use `githubToken`.

File: src/main.ts [hunk 2]
```diff
@@ -9,8 +9,9 @@ async function run(): Promise<void> {
   try {
     const repository: string = core.getInput('repository');
     const configPath: string = core.getInput('configPath');
+    const token: string = core.getInput('token');
```
Comment (kevin-hyperjump) [inline src/main.ts]: `token` to `githubToken`

File: src/utils/getConfigs.ts [hunk 3]
```diff
@@ -5,16 +5,22 @@ import { Config } from '../interfaces/Config';
 export interface GetConfig {
   repository: string;
   configPath: string;
+  token: string;
```
Comment (kevin-hyperjump) [inline src/utils/getConfigs.ts]: `token` to `githubToken`

File: src/utils/getConfigs.ts [hunk 4]
```diff
@@ -5,16 +5,22 @@ import { Config } from '../interfaces/Config';
...
 }
 
 export const getConfigs = async ({
   repository,
-  configPath
+  configPath,
+  token
 }: GetConfig): Promise<Config[]> => {
   const url = `https://api.github.com/repos/${repository}/contents/${configPath}`;
+  const hasToken = token !== '' && {
```
Comment (kevin-hyperjump) [inline src/utils/getConfigs.ts]: It's better to write ```const hasToken = token &&``` rather than ```const hasToken =  token !== ''```

Comment (kevin-hyperjump) [reply src/utils/getConfigs.ts]: Since an `undefined` will returns empty when you use ```spread object operator```.

**Review by kevin-hyperjump — CHANGES_REQUESTED** [review]

**Comment by kevin-hyperjump** [comment]
Looks good to me, but we need to make sure this feature works as expected by adding an integration test for private configs. Please update code-quality.yml to :

```yaml
name: 'Code Quality'
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  analysis-static:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2

      - name: install dependencies
        run: npm ci

      - name: run static analysis
        run: npm run static:analysis

  test-unit:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2

      - name: install dependencies
        run: npm ci

      - name: run unit test
        run: npm run test:unit

  test-integration:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2

      - name: install dependencies
        run: npm ci

      - name: build project
        run: npm run build

      - name: package project
        run: npm run package

      - name: test on public configs
      - uses: ./
        with:
          repository: 'hyperjumptech/procheck-configs'
          configPath: 'procheck-integration-test/procheck.json'

      - name: test on private configs
      - uses: ./
        with:
          repository: 'hyperjumptech/procheck-private-configs'
          configPath: 'procheck-integration-test/procheck.json'
          githubToken: $PROCHECK_GITHUB_TOKEN
        env:
          PROCHECK_GITHUB_TOKEN: ${{ secrets.PROCHECK_GITHUB_TOKEN }}
``` 

**Note** : I was created a private repository and `PROCHECK_GITHUB_TOKEN` has been defined.

**Comment by dennypradipta** [comment]
Thank you for the followups, latest commit should now include test for public and private configs :)

**Pull request closed without merging**